neural_modelling/src/neuron/plasticity/stdp/synapse_dynamics_stdp_mad_impl.c

```diff
@@ -287,6 +297,18 @@ bool synapse_dynamics_initialise(
         return false;
     }
 
+    if (sizeof(structural_rec_t)) {
+        structural_rec_array = spin1_malloc(n_neurons * sizeof(structural_rec_t));
+        if (structural_rec_array == NULL) {
+            log_error("Unable to allocate structural recording array"
+                    "- Out of DTCM");
+            return false;
+        }
+        for (uint32_t i = 0; i < n_neurons; i++) {
+        	structural_rec_array[i].recorded_val = -1;
+        }
+    }
+
```

Comment:
Yes - this is the main issue with this.  I will likely have to make this into an array in order to deal with multiple rewirings on the same timestep; it's more frequent than you might think because it's rewirings into the same post-neuron, rather than necessarily the same pair of pre- and post-neurons.

Reply:
This has moved into topographic_map_impl.c but I have not updated it yet to deal with multiple rewirings on the same timestep - that's the next step.

Reply:
I think I now have this working for multiple rewirings on a single timestep.